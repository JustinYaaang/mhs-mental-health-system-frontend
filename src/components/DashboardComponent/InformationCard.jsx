import React from 'react'
import GridItem from 'components/Grid/GridItem.jsx'
import Card from 'components/Card/Card.jsx'
import CardHeader from 'components/Card/CardHeader.jsx'
import CardIcon from 'components/Card/CardIcon.jsx'
import CardFooter from 'components/Card/CardFooter.jsx'
import DateRange from '@material-ui/icons/DateRange'
import All from '@material-ui/icons/AllInboxOutlined'

class InformationCard extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      color: '',
      category: '',
      title: '',
      value: '',
      daterange: '',
      classes: ''
    }
  }

  componentWillMount () {
    this.setState({
      color: this.props.color,
      title: this.props.title,
      value: this.props.value,
      daterange: this.props.daterange,
      classes: this.props.classes
    })
  }

  render () {
    return (
      <GridItem xs={12} sm={6} md={4}>
        <Card onClick={() => this.props.onViewClicked()}>
          <CardHeader color={this.state.color} stats icon>
            <CardIcon color={this.state.color}>
              <All />
            </CardIcon>
            <p className={this.state.classes.cardCategory}>{this.state.title}</p>
            <h3 className={this.state.classes.cardTitle}>{this.props.value}</h3>
          </CardHeader>
          <CardFooter stats>
            <div className={this.state.classes.stats}>
              <DateRange />
              {this.state.daterange}
            </div>
          </CardFooter>
        </Card>
      </GridItem>
    )
  }
}

export default InformationCard
