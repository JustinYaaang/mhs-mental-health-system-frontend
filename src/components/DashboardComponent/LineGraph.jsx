import React from 'react'
import GridItem from 'components/Grid/GridItem.jsx'
import GridContainer from 'components/Grid/GridContainer.jsx'
import Tasks from 'components/Tasks/Tasks.jsx'
import CustomTabs from 'components/CustomTabs/CustomTabs.jsx'
import Card from 'components/Card/Card.jsx'
import CardHeader from 'components/Card/CardHeader.jsx'
import CardIcon from 'components/Card/CardIcon.jsx'
import CardBody from 'components/Card/CardBody.jsx'
import CardFooter from 'components/Card/CardFooter.jsx'
import swal from 'sweetalert2'
import DateRange from '@material-ui/icons/DateRange'
import LocalOffer from '@material-ui/icons/LocalOffer'
import Update from '@material-ui/icons/Update'
import ArrowUpward from '@material-ui/icons/ArrowUpward'
import AccessTime from '@material-ui/icons/AccessTime'
import Accessibility from '@material-ui/icons/Accessibility'
import Grade from '@material-ui/icons/Grade'
import Code from '@material-ui/icons/Code'
import All from '@material-ui/icons/AllInboxOutlined'
import ChartistGraph from 'react-chartist'
class LineGraph extends React.Component {
  constructor(props){
    super(props)
  }


  render () {
    return (
      <GridItem xs={12} sm={12} md={4}>
        <Card chart>
          <CardHeader color={this.props.color}>
            <ChartistGraph
              className='ct-chart'
              data={this.props.dailySubmission}
              type={this.props.type}
              options={this.props.dashboardData.dailySalesChart.options}
              listener={this.props.dashboardData.dailySalesChart.animation}
            />
          </CardHeader>
          <CardBody>
            <h4 className={this.props.classes.cardTitle}>Daily Engagement</h4>
            <p className={this.props.classes.cardCategory}>
              <span className={this.props.classes.successText}>
                <ArrowUpward className={this.props.classes.upArrowCardCategory} />{this.props.dashboardData.percentage}%
              </span>{' '}
                  increase today.
            </p>
          </CardBody>
          <CardFooter chart>
            <div className={this.props.classes.stats}>
              <AccessTime /> updated 4 minutes ago
            </div>
          </CardFooter>
        </Card>
      </GridItem>
    )
  }
}

export default LineGraph
