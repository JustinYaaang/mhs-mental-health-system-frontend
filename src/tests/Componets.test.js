import React from 'react'
import { shallow } from 'enzyme'
import InformationCard from '../components/DashboardComponent/InformationCard'
import LineGraph from '../components/DashboardComponent/LineGraph'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

describe('Information Card', () => {
  it('should render correctly', async () => {
    const component = shallow(
      <InformationCard
        color={'info'} title={'Total Questionnaires'} value={15}
        daterange={'Updated today'} classes={{
          cardCategory: 'Dashboard-cardCategory-251',
          cardCategoryWhite: 'Dashboard-cardCategoryWhite-252',
          cardTitle: 'Dashboard-cardTitle-253',
          cardTitleWhite: 'Dashboard-cardTitleWhite-254',
          stats: 'Dashboard-stats-250',
          successText: 'Dashboard-successText-248',
          upArrowCardCategory: 'Dashboard-upArrowCardCategory-249'

        }}
      />)

    expect(component).toMatchSnapshot()
  })
})

describe('Line Graph', () => {
  it('should render correctly', async () => {
    const component = shallow(
      <LineGraph color={"success"} dailySubmission={
            {labels:[
              0: "S",
              1: "S",
              2: "M",
              3: "T",
              4: "W",
              5: "T",
              6: "F"],
              series:[
              0: 0,
              1: 8,
              2: 14,
              3: 0,
              4: 0,
              5: 0,
              6: 0
              ]
              }}
              type={"Line"}
              dashboardData={{
                dailySalesChart:{
                  options:"",
                  animation:""
                }
              }}
          classes={{
            cardCategory: 'Dashboard-cardCategory-251',
            cardCategoryWhite: 'Dashboard-cardCategoryWhite-252',
            cardTitle: 'Dashboard-cardTitle-253',
            cardTitleWhite: 'Dashboard-cardTitleWhite-254',
            stats: 'Dashboard-stats-250',
            successText: 'Dashboard-successText-248',
            upArrowCardCategory: 'Dashboard-upArrowCardCategory-249'}} 
            ></LineGraph>

        )})})