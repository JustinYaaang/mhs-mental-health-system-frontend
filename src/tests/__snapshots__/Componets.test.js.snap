// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Information Card should render correctly 1`] = `
<WithStyles(GridItem)
  md={4}
  sm={6}
  xs={12}
>
  <WithStyles(Card)>
    <WithStyles(CardHeader)
      color="info"
      icon={true}
      stats={true}
    >
      <WithStyles(CardIcon)
        color="info"
      >
        <AllInboxOutlinedIcon />
      </WithStyles(CardIcon)>
      <p
        className="Dashboard-cardCategory-251"
      >
        Total Questionnaires
      </p>
      <h3
        className="Dashboard-cardTitle-253"
      >
        15
      </h3>
    </WithStyles(CardHeader)>
    <WithStyles(CardFooter)
      stats={true}
    >
      <div
        className="Dashboard-stats-250"
      >
        <DateRangeIcon />
        Updated today
      </div>
    </WithStyles(CardFooter)>
  </WithStyles(Card)>
</WithStyles(GridItem)>
`;
